### DISCUSSION TITLE
style attribute no longer reactive

### DISCUSSION THREAD

**Discussion by nielsvh2103** [livewire/livewire#8904] [7. Bugs (archived)] [answered]
### Livewire version

v3.5.9 (latest)

### Laravel version

v11.26.0 (latest)

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Set up a simple div with a style= tag with a livewire property in there.
2. Have a button with wire click that updates that property
3. Style= is not being update, while other attrs like class are. 

See the video here;

https://github.com/user-attachments/assets/8fabce7b-c2f8-4c33-a3d2-914d07a28a5e

you can check the code in the repo but heres what i did:

```
                <button type="button"
                        wire:click="updateStyle('color:red; font-size:64px;')"
                        class="py-2.5 px-5 me-2 mb-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-100">
                    Update style
                </button>

                <button type="button"
                        wire:click="updateClass('bg-green-500 text-xl')"
                        class="py-2.5 px-5 me-2 mb-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-100">
                    Update class
                </button>

                <button type="button"
                        wire:click="resetProps"
                        class="py-2.5 px-5 me-2 mb-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-100">
                    Reset
                </button>

                <!-- this div does not update style -->
                <div class="{{ $class }}" style="{{ $style }}">
                    this div does not update style tag
                </div>

                <!-- this div does update due to wire:replace.self -->
                <div wire:replace.self class="{{ $class }}" style="{{ $style }}">
                    this div updates style tag with wire:replace.self
                </div>

                <div wire:key="{{ uniqid() }}" class="{{ $class }}" style="{{ $style }}">
                    this div updates style tag with wire:key
                </div>
```

as for the component;
```
    public string $style;
    public string $class;

    public function updateStyle($style)
    {
        $this->style = $style;
    }

    public function updateClass($class)
    {
        $this->class = $class;
    }

    public function resetProps()
    {
        $this->style = '';
        $this->class = '';
    }
```

### Repository or Wirebox URL

https://github.com/nielsvh2103/livewire-issue

### Additional information

This started happening from either 3.5.8 or 3.5.7
tested on 3.5.6 still works fine
now i have to wire key or wire:replace.self on every element that has a style tag on it where props are reactive.
its not being morphed properly it seems in the dom right now

repo has basic login with a livewire component loaded there, you can find the dummy credentials by searching for 'test@example.com'
then navigate through navigation to 'single component'
press the buttons there to reproduce.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by aigorlaxy** [comment]
I was going now to make the same topic here.

My simple code example:
`<div style="background-color:{{ $statusColor }};"> {{ $statusColor }} </div>`

While $statusColor update outside the div style attribute, the one inside doesn't.

I verified that change a days ago. I don't know exacly what happened. I tried to see if I made a change in my code, but after hours trying to "debugg", I gave up and now I'm here.

**Comment by nielsvh2103** [comment]
Update here;

I think in fact this pull request is the one that breaks it starting from 3.5.7, and i think this pr causes other issues as well, from looking at discussions/other PR :)
https://github.com/livewire/livewire/pull/8870

Workaround is to use wire:key or wire:replace.self but this is not the best course of action, as it may not work in every case.
For others having this issue i suggest locking your version at 3.5.6

**Comment by kolydart** [comment]
I am at exactly the same page. Took me a few hours of debugging to find out version 3.5.7 does not refresh `style` attribute after first load. `wire:key` does not help. I frose livewire to version 3.5.6 until the issue is resolved.

**Comment by EriksonEnterprises** [comment]
Same here. I ended up downgrading to version 3.5.6 to maintain the ability for now lol

**Comment by galengidman** [comment]
Same issue. Locked at `3.5.6` for now and that's working.

**Comment by whousehomestay** [comment, marked as answer]
Related to this https://github.com/livewire/livewire/pull/8883

**Comment by jcordingley** [comment]
Can confirm the same issue here, have downgraded to 3.5.6
